### ISSUE TITLE
Function `paginate` with SqlModel hints an Unexpected type(s) error.

### ISSUE THREAD

**Issue by eastack** [uriyyo/fastapi-pagination#1364]
![image](https://github.com/user-attachments/assets/ddeb06b9-5d66-4749-b521-a1d9fa2afb41)

There's no issue if the call chain doesn't include where.

![image](https://github.com/user-attachments/assets/7ff0e0e7-4379-455c-9f8c-33d2d9731f88)

fastapi-pagination>=0.12.32
sqlmodel>=0.0.22

**Comment by eastack** [comment]
Apologies for this issue. I’m a Python beginner and not very familiar with the Python ecosystem. Yesterday, I encountered a similar type error in Starlette. I initially thought it was due to the current limitations in type hint support within the Python ecosystem, so I opened an issue in Starlette regarding a middleware type error. However, Kludex pointed out that it might be a PyCharm issue. Eventually, after switching to VSCode, all type errors disappeared, including the ones related to your library. Everything seems fine now. I’m really sorry for wasting your time and effort. It seems you don’t need to make any changes to your code for this. 

[add_middleware hint Expected type '_MiddlewareFactory[ParamSpec("P")]', got 'Type[XXX]' instead](https://github.com/encode/starlette/issues/2760)
